apps/dgst.c

```diff
@@ -447,6 +447,11 @@ int dgst_main(int argc, char **argv)
         }
         ret = EXIT_SUCCESS;
         for (i = 0; i < argc; i++) {
+            if (argv[i] == NULL) {
```

Comment:
Under what circumstances are you finding `argv[i]` is `NULL`?

My understanding:

> The value of argc shall be nonnegative.
>
> The value argv[argc] shall be a null pointer.
>
> If the value of argc is greater than zero, the array members argv[0] through argv[argc-1] inclusive shall contain pointers to strings, which are given implementation-defined values by the host environment prior to program start up.

Reply:
I thought if the host environment does not provide arguments, then `NULL` will be sent as the argument for the file name. I can remove this `NULL` check.